core/primitives-core/src/hash.rs

```diff
@@ -24,6 +24,10 @@ impl CryptoHash {
         BorshSerialize::serialize(value, &mut hasher).unwrap();
         CryptoHash(hasher.finalize().into())
     }
+
+    pub fn as_bytes(&self) -> &[u8; 32] {
```

Comment:
Could this be an implementation of the `Deref` trait (potentially in addition to this method)?

This would allow something like this in a change above:

```rust
        self.gc_col(DBCol::Block, &block_hash);
        self.gc_col(DBCol::BlockExtra, &block_hash);
        self.gc_col(DBCol::NextBlockHashes, &block_hash);
        self.gc_col(DBCol::ChallengedBlocks, &block_hash);
        self.gc_col(DBCol::BlocksToCatchup, &block_hash);
```

Reply:
With `Deref<Target=[u8; 32]>` we need to do `&block_hash[..]` which is rather more obscure than `block_hash.as_bytes()`.  `Target=[u8]` works but feels like we’re needlessly dropping static information (size of array).  Doing Deref would also allow things like `black_hash[5]` which is rather meaningless for a CryptoHash type.

Dunon, but I’m not really convinced that Deref fits.

Reply:
Could have both, but I don't have particularly strong feelings either way.

Reply:
Let’s stick with as_bytes then for now.